xdsl/backend/riscv/register_allocation.py

```diff
@@ -169,19 +140,70 @@ def allocate_for_loop(self, loop: riscv_scf.ForOp) -> None:
             assert isinstance(yield_operand.type, RISCVRegisterType)
             assert isinstance(op_result.type, RISCVRegisterType)
 
-            if not operand.type.is_allocated:
+            if not op_result.type.is_allocated:
```

Comment:
op_result because we are walking backwards, and will already have allocated op_results used in the code below